backend/services/implementations/statisticsService.ts

```diff
@@ -0,0 +1,147 @@
+import { Types } from "mongoose";
+import MgTestSession from "../../models/testSession.model";
+import { getErrorMessage } from "../../utilities/errorUtils";
+import logger from "../../utilities/logger";
+import { ISchoolService, SchoolResponseDTO } from "../interfaces/schoolService";
+import {
+  IStatisticsService,
+  AggregateTestSessionDTO,
+  StatisticsResponseDTO,
+} from "../interfaces/statisticsService";
+import { ITestService, TestResponseDTO } from "../interfaces/testService";
+import { ITestSessionService } from "../interfaces/testSessionService";
+import IUserService from "../interfaces/userService";
+
+const Logger = logger(__filename);
+
+class StatisticsService implements IStatisticsService {
+  /* eslint-disable class-methods-use-this */
+  testSessionService: ITestSessionService;
+
+  testService: ITestService;
+
+  userService: IUserService;
+
+  schoolService: ISchoolService;
+
+  constructor(
+    testSessionService: ITestSessionService,
+    testService: ITestService,
+    userService: IUserService,
+    schoolService: ISchoolService,
+  ) {
+    this.testSessionService = testSessionService;
+    this.testService = testService;
+    this.userService = userService;
+    this.schoolService = schoolService;
+  }
+
+  async getAverageScoresBySchool(
+    testId: string,
+  ): Promise<Array<StatisticsResponseDTO>> {
+    let statisticsDTOs: Array<StatisticsResponseDTO> = [];
+
+    try {
+      const statistics: AggregateTestSessionDTO[] = await MgTestSession.aggregate(
+        [
+          // Stage 1: Filter Test Session documents by test id
+          {
+            $match: { test: Types.ObjectId(testId) },
+          },
+          // Stage 2: Group remaining documents by school id and add aggregate fields
+          {
+            $unwind: "$results",
+          },
+          {
+            $group: {
+              _id: "$school",
+              averageScore: { $avg: "$results.score" },
+              resultsPerTestArr: {
```

Comment:
Are we mapping boolean values -> 0/1 binary values here? If so, would it be more intuitive to leave this as `true/false` values here and then do a temporary conversion in `computeAverageScorePerQuestion`  instead? To avoid storing magic numbers.